src/test/java/org/jabref/gui/search/GlobalSearchBarTest.java

```diff
@@ -0,0 +1,110 @@
+import org.jabref.preferences.SearchPreferences;
+import org.jabref.testutils.category.GUITest;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.testfx.api.FxRobot;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.Start;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@GUITest
+@ExtendWith(ApplicationExtension.class)
+public class GlobalSearchBarTest {
+    private Stage stage;
+    private Scene scene;
+    private HBox hBox;
+
+    private GlobalSearchBar searchBar;
+    private StateManager stateManager;
+
+    @Start
+    public void onStart(Stage stage) {
+        SearchPreferences searchPreferences = mock(SearchPreferences.class);
+        when(searchPreferences.getSearchFlags()).thenReturn(EnumSet.noneOf(SearchRules.SearchFlags.class));
+        PreferencesService prefs = mock(PreferencesService.class, Answers.RETURNS_DEEP_STUBS);
+        when(prefs.getSearchPreferences()).thenReturn(searchPreferences);
+
+        stateManager = new StateManager();
+        // need for active database, otherwise the searchField will be disabled
+        stateManager.setActiveDatabase(new BibDatabaseContext());
+
+        // instantiate GlobalSearchBar class, so the change listener is registered
+        searchBar = new GlobalSearchBar(
+                mock(JabRefFrame.class),
+                stateManager,
+                prefs,
+                mock(CountingUndoManager.class),
+                mock(DialogService.class)
+        );
+
+        hBox = new HBox(searchBar);
+
+        scene = new Scene(hBox, 400, 400);
+        this.stage = stage;
+        stage.setScene(scene);
+
+        stage.show();
+    }
+
+    @Test
+    void recordingSearchQueriesOnFocusLostOnly(FxRobot robot) throws InterruptedException {
+        stateManager.clearSearchHistory();
+        String searchQuery = "Smith";
+        // track the node, that the search query will be typed into
+        TextInputControl searchField = robot.lookup("#searchField").queryTextInputControl();
+
+        // the focus is on searchField node, as we click on the search box
+        var searchFieldRoboto = robot.clickOn(searchField);
+        for (char c : searchQuery.toCharArray()) {
+            searchFieldRoboto.write(String.valueOf(c));
+            Thread.sleep(401);
+            assertTrue(stateManager.getWholeSearchHistory().isEmpty());
+        }
+
+        // Set the focus to another node to trigger the listener and finally record the query.
+        DefaultTaskExecutor.runInJavaFXThread(() -> hBox.requestFocus());
+        List<String> lastSearchHistory = stateManager.getWholeSearchHistory();
+
+        assertEquals(FXCollections.observableArrayList("Smith"), lastSearchHistory);
```

Comment:
Test fails:

    expected: com.sun.javafx.collections.ObservableListWrapper@34f97e7e<[Smith]> but was: com.sun.javafx.collections.ObservableListWrapper@5af934eb<[Smith]>

When playing around, I even did not get any lastSearchHistory

My current code:

```java
        List<String> lastSearchHistory = stateManager.getWholeSearchHistory().stream().toList();

        assertEquals(List.of("Smith"), lastSearchHistory);
```

Did the test ever succeed on your side?

Reply:
Both tests are successfully passed on my side

Reply:
A reasonable approach could be that  the `assertEquals` method handles the two arguments as different types of objects, something that  could cause the test to fail. However, `stateManager.getWholeSearchHistory()` also returns an observable list.

Reply:
@Siedlerchr do you face same issues?

Reply:
Last time I tried the test it was working for me as well.
Can check later this evening again